src/ChatFeed/MessageForm/index.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useState } from 'react';
+
+import { Properties } from 'csstype';
+
+import { Props } from './props';
+import { styles } from './styles';
+
+import { Attachment } from './Attachment';
+import { AttachmentInput } from './AttachmentInput';
+
+import { isImage } from '../../util/file';
+
+export const MessageForm: React.FC<Props> = ({
+  label = '',
+  style = {},
+  onChange,
+  onSubmit,
+}: Props) => {
+  const [iter, setIter] = useState(0); // Forces attachments update
+  const [value, setValue] = useState<string>('');
+  const [height, setHeight] = useState<number>(0);
+  const [buttonHover, setButtonHover] = useState<boolean>(false);
+  const [attachments, setAttachments] = useState<Array<File>>();
+
+  const overflowStyle: Properties = {
+    overflowY: height === 150 ? 'scroll' : 'hidden',
+  };
+  const buttonHoverStyle: Properties = {
+    backgroundColor: buttonHover ? '#40a9ff' : '#1890ff',
+  };
+
+  const resize = () => {
+    var textarea = document.getElementById('msg-textarea');
+    if (textarea !== null) {
+      textarea.style.height = '';
+      textarea.style.height = Math.min(textarea.scrollHeight, 150) + 'px';
+      setHeight(Math.min(textarea.scrollHeight, 150));
+    }
+  };
+
+  useEffect(() => resize(), []);
+
+  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setValue(e.target.value);
+    onChange && onChange(e);
+    resize();
+  };
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      if (value.length > 0) {
+        onSubmit && onSubmit(value);
+      }
+    }
+  };
+
+  const onRemove = (index: number) => {
+    if (attachments && attachments !== null) {
+      const newAttachments = attachments;
+      newAttachments.splice(index, 1);
+      setAttachments(newAttachments);
+      setIter(iter + 1);
```

Comment:
more of a JS question here- but does `const newAttachments = attachments;` actually do a deep copy? In the Java world `attachments` and `newAttachments` would be pointing at the same array 

if that's the case we can just simplify to this
```
setAttachments(attachments.splice(index, 1));
setIter(iter + 1);

Reply:
I'll test it out

Reply:
We gotta do this that does does not deepcopy properly